### PR TITLE
FEATURE ASH-249: Add ElastiCache, ES, Redshift reservation rights

### PR DESCRIPTION
**Pull request by kristian-lesko**
Add support for listing and purchasing reserved capacity for [ElastiCache](https://docs.aws.amazon.com/AmazonElastiCache/latest/dg/CacheNodes.Reserved.html), [OpenSearch](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/ri.html), and [Redshift](https://docs.aws.amazon.com/redshift/latest/mgmt/purchase-reserved-node-instance.html).

### CODE REVIEW

File: templates/governance.yaml [hunk 1]
```diff
@@ -273,12 +273,21 @@ Resources:
               - 'ec2:DescribeReservedInstancesOfferings'
               - 'ec2:ModifyReservedInstances'
               - 'ec2:PurchaseReservedInstancesOffering'
+              - 'elasticache:DescribeReservedCacheNodes'
+              - 'elasticache:DescribeReservedCacheNodesOfferings'
+              - 'elasticache:PurchaseReservedCacheNodesOffering'
+              - 'es:DescribeReservedInstanceOfferings'
+              - 'es:DescribeReservedInstances'
+              - 'es:PurchaseReservedInstanceOffering'
```
Comment (marko-fabry) [inline templates/governance.yaml:281]: I would also add:
[DescribeReservedElasticsearchInstanceOfferings](https://docs.aws.amazon.com/opensearch-service/latest/APIReference/API_DescribeReservedInstanceOfferings.html)
[DescribeReservedElasticsearchInstances](https://docs.aws.amazon.com/opensearch-service/latest/APIReference/API_DescribeReservedInstances.html)
[PurchaseReservedElasticsearchInstanceOffering](https://docs.aws.amazon.com/opensearch-service/latest/APIReference/API_PurchaseReservedInstanceOffering.html)

File: templates/governance.yaml [hunk 2]
```diff
@@ -273,12 +273,21 @@ Resources:
...
               - 'organizations:Describe*'
               - 'organizations:List*'
               - 'rds:DescribeDBInstances'
               - 'rds:DescribeReservedDBInstances'
               - 'rds:DescribeReservedDBInstancesOfferings'
               - 'rds:PurchaseReservedDBInstancesOffering'
+              - 'redshift:DescribeReservedNodeOfferings'
+              - 'redshift:DescribeReservedNodes'
+              - 'redshift:PurchaseReservedNodeOffering'
```
Comment (marko-fabry) [inline templates/governance.yaml:290]: I would also add:
[AcceptReservedNodeExchange](https://docs.aws.amazon.com/redshift/latest/APIReference/API_AcceptReservedNodeExchange.html)
[DescribeReservedNodeExchangeStatus](https://docs.aws.amazon.com/redshift/latest/APIReference/API_DescribeReservedNodeExchangeStatus.html)
[GetReservedNodeExchangeConfigurationOptions](https://docs.aws.amazon.com/redshift/latest/APIReference/API_GetReservedNodeExchangeConfigurationOptions.html)
[GetReservedNodeExchangeOfferings](https://docs.aws.amazon.com/redshift/latest/APIReference/API_GetReservedNodeExchangeOfferings.html)

File: templates/governance.yaml (hunk 1 above)
Comment (marko-fabry) [reply templates/governance.yaml:281]: But check with CCO team if its something they can leverage.

File: templates/governance.yaml (hunk 2 above)
Comment (marko-fabry) [reply templates/governance.yaml:290]: But check with CCO team if its something they can leverage.

**Review by marko-fabry — CHANGES_REQUESTED** [review]

File: templates/governance.yaml (hunk 1 above)
Comment (kristian-lesko) [reply templates/governance.yaml:281]: Per [IAM actions docs](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonopensearchservice.html), those three actions are deprecated and replaced by the ones added here.

**Review by marko-fabry — APPROVED** [review]

**Pull request merged**